src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatCompletion{T}.cs

```diff
@@ -125,8 +125,24 @@ public bool TryGetResult([NotNullWhen(true)] out T? result)
             return default;
         }
 
+        T? deserialized = default;
+        var wrapped = AdditionalProperties?.TryGetValue("$wrapped", out var isWrappedValue) == true && isWrappedValue is bool isWrappedBool && isWrappedBool;
+
         // If there's an exception here, we want it to propagate, since the Result property is meant to throw directly
-        var deserialized = DeserializeFirstTopLevelObject(json!, (JsonTypeInfo<T>)_serializerOptions.GetTypeInfo(typeof(T)));
+
+        if (wrapped)
+        {
+            var result = DeserializeFirstTopLevelObject(json!, (JsonTypeInfo<Payload<T>>)_serializerOptions.GetTypeInfo(typeof(Payload<T>)));
```

Comment:
This will invariably fail if the user is passing a source generated `_serializerOptions` which can't access `Payload<T>`. Assuming we decide to take this change, the result should just be parsed to a `JsonDocument` and then deserialize the nested "Data" property to the desired type.

Reply:
ah, yeah, absolutely!

Reply:
Would this fail equally?

```
            schemaNode = (JsonObject)serializerOptions.GetJsonSchemaAsNode(typeof(Payload<T>), exporterOptions);
```

🤔